src/main/scala/viper/gobra/frontend/ParseTreeTranslator.scala

```diff
@@ -0,0 +1,2271 @@
+    visitChildren(ctx) match {
+      // Declaration statements are wrapped in sequences to constitute a statement
+      case decl : Vector[PDeclaration] @unchecked => PSeq(decl).at(ctx)
+      case s : PStatement => s
+      case _ => unexpected(ctx)
+    }
+  }
+
+
+  /**
+    * {@inheritDoc  }
+    *
+    * <p>The default implementation returns the result of calling
+    * {@link #   visitChildren} on {@code ctx}.</p>
+    */
+  override def visitEmptyStmt(ctx: EmptyStmtContext): PEmptyStmt = PEmptyStmt().at(ctx)
+
+  //region Labeled statements
+  /**
+    * {@inheritDoc  }
+    *
+    * <p>The default implementation returns the result of calling
+    * {@link #   visitChildren} on {@code ctx}.</p>
+    */
+  override def visitLabeledStmt(ctx: LabeledStmtContext): PLabeledStmt = {
+    val label = visitLabelDef(ctx.IDENTIFIER())
+    val stmt = visitNodeOrElse[PStatement](ctx.statement(), PEmptyStmt().at(label))
+    PLabeledStmt(label, stmt).at(ctx)
+  }
+
+  def visitLabelDef(node: TerminalNode) : PLabelDef = {
+    PLabelDef(node.toString).at(node)
+  }
+
+  def visitLabelUse(node: TerminalNode) : PLabelUse = {
+    PLabelUse(node.toString).at(node)
+  }
+  //endregion
+
+  //region Expression statements
+  /**
+    * {@inheritDoc  }
+    *
+    * <p>The default implementation returns the result of calling
+    * {@link #   visitChildren} on {@code ctx}.</p>
+    */
+  override def visitExpressionStmt(ctx: ExpressionStmtContext): PExpressionStmt = PExpressionStmt(visitNode[PExpression](ctx.expression())).at(ctx)
+  //endregion
+
+  //region Send statements
+  /**
+    * {@inheritDoc  }
+    *
+    * <p>The default implementation returns the result of calling
+    * {@link #   visitChildren} on {@code ctx}.</p>
+    */
+  override def visitSendStmt(ctx: SendStmtContext): PSendStmt = {
+    visitChildren(ctx) match {
+      case Vector(channel : PExpression, "<-", msg : PExpression) => PSendStmt(channel, msg).at(ctx)
+    }
+  }
+  //endregion
+
+  //region IncDec statements
+  /**
+    * {@inheritDoc  }
+    *
+    * <p>The default implementation returns the result of calling
+    * {@link #visitChildren} on {@code ctx}.</p>
+    */
+  override def visitIncDecStmt(ctx: IncDecStmtContext): PAssignmentWithOp = {
+    val exp = visitNode[PExpression](ctx.expression()) match {
+      case assignee : PAssignee => assignee
+      case _ => fail(ctx.expression(), "Increment/Decrement-statements must have an assignee as operand.")
+    }
+    if (has(ctx.PLUS_PLUS()))
+      PAssignmentWithOp(PIntLit(1).at(ctx.PLUS_PLUS()), PAddOp().at(ctx.PLUS_PLUS()), exp).at(ctx) else
+      PAssignmentWithOp(PIntLit(1).at(ctx.MINUS_MINUS()), PSubOp().at(ctx.MINUS_MINUS()), exp).at(ctx)
+  }
+  //endregion
+
+  //region Assignments
+  /**
+    * Visit a parse tree produced by `GobraParser`.
+    *
+    * @param ctx the parse tree
+    * @return the visitor result
+    */
+  // TODO : Refactor easy
```

Comment:
there are still a couple of refactoring TODOs in this file. What do you plan to do with them? Should they still be addressed as part of this PR or do you want to do them after merging?

Reply:
push

Reply:
Most of these would be nice to have, but are readable as is. It may be best to leave them for now.

Reply:
ok.